build.sbt

```diff
@@ -0,0 +1,149 @@
+ * comments the less obvious settings.
+ *
+ * This nicely leads me to explaning goal and non-goals of this build definition. Goals are:
+ *
+ *   - to be easy to tweak it in case a bug or small inconsistency is found
+ *   - to mimic Ant's behavior as closely as possible
+ *   - to be super explicit about any departure from standard sbt settings
+ *   - to achieve functional parity with Ant build as quickly as possible
+ *   - to be readable and not necessarily succint
+ *   - to provide the nicest development experience for people hacking on Scala
+ *
+ * Non-goals are:
+ *
+ *   - to have the shortest sbt build definition possible; we'll beat Ant definition
+ *     easily and that will thrill us already
+ *   - to remove irregularities from our build process right away
+ *   - to modularize the Scala compiler or library further
+ *
+ * It boils down to simple rules:
+ *
+ *   - project laytout is set in stone for now
+ *   - if you need to work on convincing sbt to follow non-standard layout then
+ *     explain everything you did in comments
+ *   - constantly check where Ant build produces class files, artifacts, what kind of other
+ *     files generates and port all of that to here
+ *
+ * Note on bootstrapping:
+ *
+ *   Let's start with reminder what bootstrapping means in our context. It's an answer
+ *   to this question: which version of Scala are using to compile Scala? The fact that
+ *   the question sounds circular suggests trickiness. Indeed, bootstrapping Scala
+ *   compiler is a tricky process.
+ *
+ *   Ant build used to have involved system of bootstrapping Scala. It would consist of
+ *   three layers: starr, locker and quick. The sbt build for Scala ditches layering
+ *   and strives to be as standard sbt project as possible. This means that we are simply
+ *   building Scala with latest stable release of Scala.
+ *   See this discussion for more details behind this decision:
+ *     https://groups.google.com/d/topic/scala-internals/gp5JsM1E0Fo/discussion
+ */
+
+lazy val commonSettings = Seq(
+  organization := "org.scala-lang",
+  version := "2.11.6-SNAPSHOT",
+  scalaVersion := "2.11.5",
+  // all project will have baseDirectory set to root folder; we shouldn't include
+  // any source from it in any project
+  sourcesInBase := false,
+  // we don't cross build Scala itself
+  crossPaths := false,
+  // do not add Scala library jar as a dependency automatically
+  autoScalaLibrary := false,
+  // we always assume that Java classes are standalone and do not have any dependency
+  // on Scala classes
+  compileOrder := CompileOrder.JavaThenScala,
+  // we don't want any unmanaged jars; as a reminder: unmanaged jar is a jar stored
+  // directly on the file system and it's not resolved through Ivy
+  // Ant's build stored unmanaged jars in `lib/` directory
+  unmanagedJars in Compile := Seq.empty,
+  // set baseDirectory to the root folder in all projects
+  baseDirectory := (baseDirectory in ThisBuild).value,
```

Comment:
This will likely cause significant issues for IDEs. They definitely want each project to live in a different directory, don't they? Or is `baseDirectory` not use by, say, `sbteclipse`?